extras/kafka.connect/api/src/main/java/org/apache/rya/kafka/connect/api/sink/RyaSinkTask.java

```diff
@@ -0,0 +1,144 @@
+ * <p/>
+ * Implementations of this class only need to specify functionality that is specific to the
+ * Rya implementation.
+ */
+@DefaultAnnotation(NonNull.class)
+public abstract class RyaSinkTask extends SinkTask {
+    private static final Logger log = LoggerFactory.getLogger(RyaSinkTask.class);
+
+    @Nullable
+    private SailRepository sailRepo = null;
+
+    @Nullable
+    private SailRepositoryConnection conn = null;
+
+    /**
+     * Throws an exception if the configured Rya Instance is not already installed
+     * within the configured database.
+     *
+     * @param taskConfig - The configuration values that were provided to the task. (not null)
+     * @throws ConnectException The configured Rya Instance is not installed to the configured database
+     *   or we were unable to figure out if it is installed.
+     */
+    protected abstract void checkRyaInstanceExists(final Map<String, String> taskConfig) throws ConnectException;
+
+    /**
+     * Creates an initialized {@link Sail} object that may be used to write {@link Statement}s to the configured
+     * Rya Instance.
+     *
+     * @param taskConfig - Configures how the Sail object will be created. (not null)
+     * @return The created Sail object.
+     * @throws ConnectException The Sail object could not be made.
+     */
+    protected abstract Sail makeSail(final Map<String, String> taskConfig) throws ConnectException;
+
+    @Override
+    public String version() {
+        return Manifests.exists("Build-Version") ? Manifests.read("Build-Version"): "UNKNOWN";
+    }
+
+    @Override
+    public void start(final Map<String, String> props) throws ConnectException {
+        requireNonNull(props);
+
+        // Ensure the configured Rya Instance is installed within the configured database.
+        checkRyaInstanceExists(props);
+
+        // Create the Sail object that is connected to the Rya Instance.
+        final Sail sail = makeSail(props);
+        sailRepo = new SailRepository( sail );
+        conn = sailRepo.getConnection();
+    }
+
+    @Override
+    public void put(final Collection<SinkRecord> records) {
+        requireNonNull(records);
+
+        // Return immediately if there are no records to handle.
+        if(records.isEmpty()) {
+            return;
+        }
+
+        // If a transaction has not been started yet, then start one.
+        if(!conn.isActive()) {
+            conn.begin();
+        }
+
+        // Iterate through the records and write them to the Sail object.
+        for(final SinkRecord record : records) {
+            // If everything has been configured correctly, then the record's value will be a Set<Statement>.
+            conn.add((Set<? extends Statement>) record.value());
+        }
+    }
+
+    @Override
+    public void flush(final Map<TopicPartition, OffsetAndMetadata> currentOffets) {
+        // Flush the current transaction.
+        conn.commit();
+    }
+
+    @Override
+    public void stop() {
+        try {
```

Comment:
what about the actual Sail that is used?  are you expecting that to be shutdown by whatever uses this?

Reply:
that sail object is shut down internally within the SailRepository class when the SailRepository's shutdown method is invoked. Check like 169 of SailRepository.

Reply:
👍